From 8b21dcb3b1f4439fce6e0153b96354dcea01971a Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Fri, 22 Apr 2022 22:35:18 +0200
Subject: [PATCH 02/22] winex11.drv: Call NtUserSetForegroundWindow
 instead of NtUserSetActiveWindow on restore.

So that the window is both active and foreground before we send the
SC_RESTORE command. Project Cars 3 expects that as it tries to reacquire
DInput devices on SC_RESTORE.

CW-Bug-Id: #19011
CW-Bug-Id: #20227
---
 dlls/winex11.drv/event.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/winex11.drv/event.c b/dlls/winex11.drv/event.c
index 11111111111..11111111111 100644
--- a/dlls/winex11.drv/event.c
+++ b/dlls/winex11.drv/event.c
@@ -1341,7 +1341,7 @@ static void handle_wm_state_notify( HWND hwnd, XPropertyEvent *event, BOOL updat
                 TRACE( "restoring win %p/%lx\n", data->hwnd, data->whole_window );
                 release_win_data( data );
                 if ((style & (WS_MINIMIZE | WS_VISIBLE)) == (WS_MINIMIZE | WS_VISIBLE))
-                    NtUserSetActiveWindow( hwnd );
+                    NtUserSetForegroundWindow( hwnd, FALSE );
                 send_message( hwnd, WM_SYSCOMMAND, SC_RESTORE, 0 );
                 return;
             }
-- 
2.36.0

